supreme/src/androidMain/kotlin/at/asitplus/signum/supreme/agreement/KeyAgreement.android.kt

```diff
@@ -0,0 +1,45 @@
+package at.asitplus.signum.supreme.agreement
+
+import android.security.keystore.UserNotAuthenticatedException
+import at.asitplus.signum.indispensable.CryptoPublicKey
+import at.asitplus.signum.indispensable.toJcaPublicKey
+import at.asitplus.signum.supreme.HazardousMaterials
+import at.asitplus.signum.supreme.dsl.DSL
+import at.asitplus.signum.supreme.dsl.DSLConfigureFn
+import at.asitplus.signum.supreme.hazmat.jcaPrivateKey
+import at.asitplus.signum.supreme.os.AndroidKeystoreSigner
+import at.asitplus.signum.supreme.os.AndroidSignerSigningConfiguration
+import at.asitplus.signum.supreme.os.PlatformSigningProviderSignerSigningConfigurationBase
+import at.asitplus.signum.supreme.sign.Signer
+
+actual suspend fun Signer.ECDSA.performAgreement(
```

Comment:
why is this an expect/actual extension function and not part of the signer interface (implemented by the existing platform classes?)